File: Data Structure+Algorithm/Searching/Beginner/Binary Search/binary_search.c
#include <stdio.h>

int binarysearch(int key, int a[], int size) {
    int i, low, mid, high;
    low = 0;
    high = size - 1;
    while(low <= high){
        mid = (low + high) / 2;
        if(a[mid] == key)
            return (mid + 1);
        else if(a[mid] > key)
            high = mid - 1;
        else
            low = mid + 1;
    }

    return 0;
}

int main() {
    int key, i, pos, size, a[20];
    printf("Enter the size: ");
    scanf("%d", &size);
    printf("enter the list: ");
    for(i = 0; i < size; i++) {
        scanf("%d", &a[i]);
    }
    printf("Enter the element to be found: ");
    scanf("%d", &key);
    pos = binarysearch(key, a, size);
    if(pos == 0)
        printf("element is not found\n");
    else
        printf("key %d is found at %d\n", key, pos);
    return 0;
}

File: Data Structure+Algorithm/Searching/Beginner/Binary Search/BinarySearch.java
//Binary Search in Java demonstrated using function call

class BinarySearchExample{  
 public static void binarySearch(int arr[], int first, int last, int key){  
   int mid = (first + last)/2;                                                 //Splitting array in half here.
   while( first <= last ){  
      if ( arr[mid] < key ){  
        first = mid + 1;     
      }else if ( arr[mid] == key ){                                           //Evaluates to TRUE if element is found at mid.
        System.out.println("Element is found at position: " + mid);  
        return;  
      }else{  
         last = mid - 1;  
      }  
      mid = (first + last)/2;  
   }  
   System.out.println("Element is not found!");  
 }  
 public static void main(String args[]){  
        int arr[] = {10,20,30,40,50};  
        int key = 30;  
        int last=arr.length-1;  
        binarySearch(arr,0,last,key);     
 }  
}
